client/packages/executor/src/executionManager/sideEffect.ts

```diff
@@ -173,8 +175,8 @@ export class SideEffect extends EventEmitter {
             this.target = new TextDecoder().decode(sideEffect.target.toU8a())
             this.gateway = sdk.gateways[this.target]
             this.securityLevel = this.evalSecurityLevel(this.gateway.gatewayType)
-            this.reward = new BehaviorSubject(sdk.circuit.toFloat(sideEffect.maxReward)) // this is always in TRN (native asset)
-            this.insurance = sdk.circuit.toFloat(sideEffect.insurance) // this is always in TRN (native asset)
+            this.reward = new BehaviorSubject(new BN.BN(sdk.circuit.toFloat(sideEffect.maxReward))) // this is always in TRN (native asset)
```

Comment:
It doesnt make a lot of sense to convert to float, and then to BN, we essentially loose all precision. BN supports no floats.